### PR TITLE
[Metricbeat] Add billing metricset into googlecloud module

### PR DESCRIPTION
**Pull request by kaiyan-sheng**
## What does this PR do?

This PR is to add billing metricset into googlecloud module. For GCP, there is no direct billing API like AWS Cost Explorer to query for billing information. In order to get the billing date, users need to export billing reports into GCP BigQuery database and then this billing metricset can make some queries to get the actual billing data out.

## Why is it important?

This will make GCP users' life a lot easier when tracking billing data 😄 

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

It took me a while to get permissions for GCP billing account. Not sure how this will be tested in a short period of time. The config I used for billing metricset is:
```
- module: googlecloud
  metricsets:
    - billing
  project_id: "elastic-bi"
  credentials_file_path: "/Users/kaiyansheng/Desktop/elastic-bi-gcp.json"
  dataset_id: "master_gcp"
  table_pattern: "gcp_billing_export_v1"
  cost_type: "regular"
  period: 24h
```

## Related issues
- Closes https://github.com/elastic/beats/issues/20738

## Screenshots
Screen Shot 2020-09-01 at 1 53 06 PM (2)

### CODE REVIEW

File: x-pack/metricbeat/module/googlecloud/billing/_meta/data.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    },
+    "event": {
+        "dataset": "googlecloud.billing",
+        "duration": 115000,
+        "module": "googlecloud"
+    },
+    "googlecloud": {
+        "billing": {
+            "cost_type": "regular",
+            "invoice_month": "202008",
```
Comment (exekias) [inline x-pack/metricbeat/module/googlecloud/billing/_meta/data.json:18]: Would it make sense to convert this to a timestamp?

File: x-pack/metricbeat/module/googlecloud/billing/billing.go [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+	event.RootFields = common.MapStr{}
+
+	event.RootFields.Put("cloud.provider", "googlecloud")
+	event.RootFields.Put("cloud.account.id", accountID)
+	event.RootFields.Put("cloud.account.name", accountID)
+
+	event.MetricSetFields.Put("invoice_month", rowItems[0])
+	event.MetricSetFields.Put("project_id", rowItems[1])
+	event.MetricSetFields.Put("cost_type", rowItems[2])
+	event.MetricSetFields.Put("total", rowItems[3])
```
Comment (exekias) [inline x-pack/metricbeat/module/googlecloud/billing/billing.go:252]: nit: you could initialize all this together, no need to do a `Put` call right?

File: x-pack/metricbeat/module/googlecloud/billing/_meta/docs.asciidoc [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+----
+- module: googlecloud
+  metricsets:
+    - billing
+  period: 24h
+  project_id: "your project id"
+  credentials_file_path: "your JSON credentials file path"
+  dataset_id: "dataset id"
+  table_pattern: "table pattern"
+  cost_type: "regular"
```
Comment (exekias) [inline x-pack/metricbeat/module/googlecloud/billing/_meta/docs.asciidoc:23]: I miss a few explanations on what these parameters are. I'm personally not very familiar with this service so I don't fully understand what to put here

**Review by exekias — CHANGES_REQUESTED** [review]
This will be a great addition! I left a few minor comments

File: x-pack/metricbeat/module/googlecloud/billing/billing.go (hunk 2 above)
Comment (kaiyan-sheng) [reply x-pack/metricbeat/module/googlecloud/billing/billing.go:252]: haha not sure why I had to use `Put`... Will fix it in the next commit!! Thank you!!

File: x-pack/metricbeat/module/googlecloud/billing/_meta/docs.asciidoc (hunk 3 above)
Comment (kaiyan-sheng) [reply x-pack/metricbeat/module/googlecloud/billing/_meta/docs.asciidoc:38]: Done. Thank you!

File: x-pack/metricbeat/module/googlecloud/billing/_meta/data.json (hunk 1 above)
Comment (kaiyan-sheng) [reply x-pack/metricbeat/module/googlecloud/billing/_meta/data.json:18]: Hmm I think we should keep it as a string to match `invoice.month` in billing table schema from BigQuery: https://cloud.google.com/billing/docs/how-to/export-data-bigquery-tables#data-schema

**Review by exekias — APPROVED** [review]

**Comment by kaiyan-sheng** [comment]
CI failed with `Build and Test / Filebeat oss / filebeat.tests.system.test_autodiscover.TestAutodiscover.test_default_settings`, not related to this PR.

**Pull request merged**